src/services/getEditsForFileRename.ts

```diff
@@ -1,65 +1,128 @@
-        const pathUpdater = getPathUpdater(oldFilePath, newFilePath, host);
+    export function getEditsForFileRename(program: Program, oldFileOrDirPath: string, newFileOrDirPath: string, host: LanguageServiceHost, formatContext: formatting.FormatContext, preferences: UserPreferences): ReadonlyArray<FileTextChanges> {
+        const oldToNew = getPathUpdater(oldFileOrDirPath, newFileOrDirPath);
+        const newToOld = getPathUpdater(newFileOrDirPath, oldFileOrDirPath);
         return textChanges.ChangeTracker.with({ host, formatContext }, changeTracker => {
-            updateTsconfigFiles(program, changeTracker, oldFilePath, newFilePath);
-            for (const { sourceFile, toUpdate } of getImportsToUpdate(program, oldFilePath, host)) {
-                const newPath = pathUpdater(isRef(toUpdate) ? toUpdate.fileName : toUpdate.text);
-                if (newPath !== undefined) {
-                    const range = isRef(toUpdate) ? toUpdate : createStringRange(toUpdate, sourceFile);
-                    changeTracker.replaceRangeWithText(sourceFile, range, isRef(toUpdate) ? newPath : removeFileExtension(newPath));
-                }
-            }
+            updateTsconfigFiles(program, changeTracker, oldToNew);
+            updateImports(program, changeTracker, oldToNew, newToOld, host, preferences);
         });
     }
 
-    function updateTsconfigFiles(program: Program, changeTracker: textChanges.ChangeTracker, oldFilePath: string, newFilePath: string): void {
-        const configFile = program.getCompilerOptions().configFile;
+    /** If 'path' refers to an old directory, returns path in the new directory. */
+    type PathUpdater = (path: string) => string | undefined;
+    function getPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string): PathUpdater {
+        return path => {
+            if (path === oldFileOrDirPath) return newFileOrDirPath;
+            const suffix = tryRemovePrefix(path, oldFileOrDirPath);
+            return suffix === undefined ? undefined : newFileOrDirPath + suffix;
+        };
+    }
+
+    function updateTsconfigFiles(program: Program, changeTracker: textChanges.ChangeTracker, oldToNew: PathUpdater): void {
+        const { configFile } = program.getCompilerOptions();
         if (!configFile) return;
-        const oldFile = getTsConfigPropArrayElementValue(configFile, "files", oldFilePath);
-        if (oldFile) {
-            changeTracker.replaceRangeWithText(configFile, createStringRange(oldFile, configFile), newFilePath);
+        const jsonObjectLiteral = getTsConfigObjectLiteralExpression(configFile);
+        if (!jsonObjectLiteral) return;
+
+        for (const property of jsonObjectLiteral.properties) {
+            if (!isPropertyAssignment(property) || !isStringLiteral(property.name)) continue;
+            switch (property.name.text) {
+                case "files":
+                case "include":

```

Comment:
include could be "src/*.ts" and your old path will be oldFileOrDirPath  will be full Path correct? (eg. /user/username/project/src). Don't you want to handle that?

Reply:
If you are renaming file from "/projects/src/file1.ts" to "/projects/lib/file1.ts" and include is "src/*.ts" do you want to add "lib/file1.ts" in the include section ?